Date/Date.cpp

```diff
@@ -0,0 +1,63 @@
+#include <iostream>
+#include "Date.h"
+
+using namespace std;
+
+void Date::validate(int day, int month, int year) {
+    int daysInMonth[13] = {1, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
+
+    if ( day > daysInMonth[month] || day < daysInMonth[0]) {
+        throw InvalidDateException("Invalid Day!");
+    }
+    if ( month < 1 || month > 12 ) {
+        throw InvalidDateException("Invalid Month!");
+    }
+    if ( year < 0 ) {
+        throw InvalidDateException("Invalid Year!");
+    }
+    if ( month == 2 || month == 4 || month == 6 || month == 9 || month == 11 ) {
+        if ( day == 31) {
+            throw InvalidDateException("Invalid Month!");
+        }
+    }
```

Comment:
Come on =(
`if ( day > daysInMonth[month] || day < daysInMonth[0]) {` is not enough for you?